### PR TITLE
CMP1903 A1 Submission

### PR DESCRIPTION
**Pull request by ThomGitHub**

### CODE REVIEW

File: CMP1903_A1_2324/Program.cs [hunk 1]
```diff
@@ -6,37 +6,46 @@
...
+            {
+                Environment.Exit(0);
+                // Terminates the program.
+            } 
+            else if (userInput != "E" && userInput != "T" && userInput != "R")
+            {
+                Console.WriteLine("The only valid inputs are 'R' and 'E'.");
+                // If the user makes an incorrect input, sends an error message.
+                Game.values.Clear();
+            }
```
Comment (Cait43) [inline CMP1903_A1_2324/Program.cs:48]: You could added T as an option in the error message

File: CMP1903_A1_2324/Testing.cs [hunk 2]
```diff
@@ -8,13 +9,30 @@ namespace CMP1903_A1_2324
...
+            Console.WriteLine("Running test..."); 
+            testGame.RollDice(); 
+
+            foreach (int i in Game.values)
+            {
+                Debug.Assert(i >= 1, "Die has rolled below a 1.");
+                Debug.Assert(i <= 6, "Die has rolled above a 6.");
+                // Checks each of the dice rolled and whether or not they're above 6 or below 1.
+            }
+
```
Comment (Cait43) [inline CMP1903_A1_2324/Testing.cs:28]: Check that you have < and > the right way around for the error message you're showing

**Review by Cait43 — COMMENTED** [review]
It would make the code easier to understand if you added more comments explaining what your code is doing specifically in the Game class.

Error handling for out-of-bounds is present, with additional error handling for incorrect input of re-roll. However, checks of the error messages shown should be made for incorrect wording. 

Most of the requirements appear to be met; however, you seem to be missing checks for whether the sum is within the bounds 3 - 18.

Code appears to follow all the formatting guidelines set, would suggest capitalizing values because it's public.

**Comment by inkpin** [comment]
The game clearly follows the brief of the assessment, the dice rolls are completed successfully and their is error checking present. 
The code is easy to read and the functionality is well developed. 

A potential improvement is to verify the names of your variables and ensure they're accurately named, for example: newGame could be named newInstance, providing a clear understanding to individuals reading through your code :) 

ave the requirements been met: **Yes**
Is the code formatted using the Style Guidelines correctly: **Yes**
Is the code easy to read: **Yes**
Are different errors handled correctly: **Yes**

**Pull request merged**